File: Golang/Facade/Beginner/subsystems/sound_system.go
package subsystems

import (
	"fmt"
)

type SoundSystem struct {}

func (s *SoundSystem) TurnOn() {
	fmt.Println("Turning on sound system")
}

func (s *SoundSystem) TurnOff() {
	fmt.Println("Turning of sound system")
}

func (s *SoundSystem) SetVolume(volume int32) {
	fmt.Printf("The volume set is %d\n", volume)
}

File: Java/Facade/MovieStreamFacade/src/subsystem/SoundSystem.java
package subsystem;

public class SoundSystem {
    public void turnOn() {
        System.out.println("Sound system is ON");
    }

    public void turnOff() {
        System.out.println("Sound system is OFF");
    }

    public void setVolume(int level) {
        System.out.println(
                "Volume set to " + level
        );
    }
}
